airbyte-cdk/python/airbyte_cdk/sources/declarative/decoders/pagination_decoder_decorator.py

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import logging
+from typing import Any, Generator, MutableMapping
+
+import requests
+from airbyte_cdk.sources.declarative.decoders import Decoder
+
+logger = logging.getLogger("airbyte")
+
+
+class PaginationDecoderDecorator(Decoder):
```

Comment:
Is it intentional that we are not using @dataclass for this implementation?

Reply:
Nope! good catch. Thanks